src/main/groovy/uk/ac/sanger/scgcf/jira/lims/post_functions/SubmissionPostFunctions.groovy

```diff
@@ -0,0 +1,114 @@
+                if(plateIssue.getStatus().getName() == 'PltSS2 Rdy for Submission') {
+
+                    // fetch action id for this transition from config file
+                    def mainConfigKey = "transitions"
+                    def workflowName = "plate_ss2"
+                    def transitionAlias = "START_SUBMISSION"
+                    int actionId = ConfigReader.getConfigElement([mainConfigKey, workflowName, transitionAlias, "tactionid"]) as int
+
+                    // transition the issue to 'In Submission'
+                    ErrorCollection ec = WorkflowUtils.transitionIssue(plateIssue, actionId)
+                    if(ec != null) {
+                        LOG.error("Error transitioning plate issue with ID ${plateIdString}".toString())
+                        // Get all non field-specific error messages
+                        Collection<String> stringErrors = ec.getErrorMessages()
+                        stringErrors.eachWithIndex{ String err, int i ->
+                            LOG.error("Error ${i}: ${err}".toString())
+                        }
+                    }
+                }
+            } else {
+                LOG.error("Plate issue with ID ${plateIdString} is not valid when adding to Submission with ID ${submIssue.id}".toString())
+            }
+        }
+    }
+
+    /**
+     * Remove the links between a list of plates and the specified Submission issue and transition them if appropriate.
+     *
+     * @param arrayPlateIds the list of plate issue ids
+     * @param submIssue the Submission issue
+     */
+    public static void removePlatesFromSubmission(String[] arrayPlateIds, Issue submIssue) {
+
+        // for each issue in list de-link it from the Submission issue and check if state should be changed
+        arrayPlateIds.each { String plateIdString ->
+            Long plateIdLong = Long.parseLong(plateIdString)
+            LOG.debug("Removing link to plate with ID ${plateIdString} from Submission".toString())
+
+            def issMngr = ComponentAccessor.getIssueManager()
+
+            def plateIssue = issMngr.getIssueObject(plateIdLong)
+            def isPlateIssueValid = (plateIssue != null) && (plateIssue.getIssueType().getName() == 'Plate SS2')
+
+            if(isPlateIssueValid) {
+
+                // remove the link between the two issues
+                WorkflowUtils.removeIssueLink(submIssue, plateIssue, 'Group includes')
+
+                // if state of plate was 'In Submission' transition the issue back to 'Rdy for Submission'
+                if(plateIssue.getStatus().getName() == 'PltSS2 In Submission') {
+
+                    def mainConfigKey = "transitions"
+                    def workflowName = "plate_ss2"
+                    def transitionAlias = "REVERT_TO_READY_FOR_SUBMISSION"
+                    int actionId = ConfigReader.getConfigElement([mainConfigKey, workflowName, transitionAlias, "tactionid"]) as int
```

Comment:
From line 97-107 is a duplication of 49-60. You could extract it to a private method and call that one for error handling.